src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -2109,6 +2109,43 @@ public FixableDiagnostic ProviderDeclarationViaImportKeywordIsDeprecated(Provide
                 TextSpan,
                 "BCP386",
                 $@"The decorator ""{decoratorName}"" may not be used on statements whose declared type is a reference to a resource-derived type.");
+
+            public ErrorDiagnostic NegatedTypeIndexSought() => new(
+                TextSpan,
+                "BCP387",
+                "Indexing into a type requires a natural number or 0. Unary operators are not permitted.");
+
+            public ErrorDiagnostic TupleRequiredForIndexAccess(TypeSymbol wrongType) => new(
+                TextSpan,
+                "BCP388",
+                $"Cannot access elements of type \"{wrongType}\" by index. An tuple type is required.");
+
+            public ErrorDiagnostic ExplicitAdditionalPropertiesTypeRequiredForAccessThereto(TypeSymbol wrongType) => new(
+                TextSpan,
+                "BCP389",
+                $"The type \"{wrongType}\" does not declare an additional properties type.");
+
+            public ErrorDiagnostic ExplicitItemsTypeRequiredForAccessThereto(TypeSymbol wrongType) => new(
+                TextSpan,
+                "BCP390",
+                $"The type \"{wrongType}\" does not declare an element type.");
+
+            public ErrorDiagnostic AccessExpressionForbiddenBase() => new(
+                TextSpan,
+                "BCP391",
+                "Access expressions within type clauses are only allowed if the innermost base expression is a symbolic reference to a type.");
+
+            public Diagnostic InvalidResourceTypeIdentifier(string resourceTypeIdentifier) => new(
```

Comment:
I can't find any references to code "BCP388", "BCP391" & "BCP392" - could you make sure we have coverage (e.g. to verify the diagnostic is raised and formatted correctly)? The simplest option would be to just add some cases to the baseline tests.